### PR TITLE
Feat ✨ (#8159)Implement Web Analytics logic from frontend

### LINKED ISSUE #8159: Implement Web Analytics logic from frontend
Reference : #8075 

As part of 0.13.0, we are planning to build web analytics for the OpenMetadata Platform to track user activity and data entities. 
To track user activity like what page user has visited and how much time they are spending on each page FronEnd has to build a script that will collect these data and send it to the backend to store the user activities.

### PR DESCRIPTION
**Pull request by Sachin-chaurasiya**
Fixes #8159 
### Describe your changes :

I worked on adding web analytics logic on the frontend to capture the page views.

#
### Type of change :

- [x] New feature

#
### Frontend Preview (Screenshots) :

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
@open-metadata/ui

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/WebAnalytics/WebAnalyticsProvider.test.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+};
+
+describe('Test WebAnalytics Component', () => {
+  it('Should render the child component', async () => {
+    await act(async () => {
+      render(<WebAnalyticsProvider {...mockProps} />, {
+        wrapper: MemoryRouter,
+      });
+    });
+
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/WebAnalytics/WebAnalyticsProvider.test.tsx:29]:
```suggestion

describe('Test WebAnalytics Component', () => {
  it('Should render the child component', async () => {
    await act(async () => {
      render(<WebAnalyticsProvider><div data-testid="mock-children">Mock Children</div></WebAnalyticsProvider>, {
        wrapper: MemoryRouter,
      });
    });

```

File: openmetadata-ui/src/main/resources/ui/src/WebAnalytics/helper.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        hostname,
+        language: navigator.language,
+        screenSize: `${properties.width}x${properties.height}`,
+        userId,
+        sessionId: anonymousId,
+        referrer: properties.referrer,
+      };
+
+      // eslint-disable-next-line
+      console.log(eventData);
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/WebAnalytics/helper.ts:56]: is it intentional? if yes, is there any reason for the same?

Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/WebAnalytics/helper.ts:56]: Please read the comments, we don't have the backend API support so just added the console log.

File: openmetadata-ui/src/main/resources/ui/src/WebAnalytics/WebAnalyticsProvider.test.tsx (hunk 1 above)
Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/WebAnalytics/WebAnalyticsProvider.test.tsx:29]: Is there any difference between the one and suggested changes?

Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/WebAnalytics/WebAnalyticsProvider.test.tsx:29]: with that, we can check the same behaviour as App

File: openmetadata-ui/src/main/resources/ui/src/WebAnalytics/helper.ts (hunk 2 above)
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/WebAnalytics/helper.ts:57]: Yes, comments were not added with console statement earlier, so missed that, Thanks for making it together with the console 😁

**Comment by Sachin-chaurasiya** [comment]
Thanks, @ShaileshParmar11, All comments are addressed.

**Review by ShaileshParmar11 — DISMISSED** [review]
LGTM!

File: openmetadata-ui/src/main/resources/ui/src/WebAnalytics/WebAnalyticsProvider.test.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        <WebAnalyticsProvider>
+          <div data-testid="mock-children">Mock Children</div>
+        </WebAnalyticsProvider>,
+        {
+          wrapper: MemoryRouter,
+        }
+      );
+    });
+
+    const children = await screen.findByTestId('mock-children');
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/WebAnalytics/WebAnalyticsProvider.test.tsx:32]: This test isn't making any sense. What we are trying to test here?

File: openmetadata-ui/src/main/resources/ui/src/WebAnalytics/WebAnalyticsProvider.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+import { observer } from 'mobx-react';
+import React, { ReactNode, useMemo } from 'react';
+import { AnalyticsProvider } from 'use-analytics';
+import AppState from '../AppState';
+import { getAnalyticInstance } from './helper';
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/WebAnalytics/WebAnalyticsProvider.tsx:18]: Let's call it util to be consistence with our folder structure?

File: openmetadata-ui/src/main/resources/ui/src/WebAnalytics/helper.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+/**
+ * track the page view if user id is available.
+ * @param _pageData PageData
+ * @param userId string
+ */
+export const trackPageView = (pageData: WebPageData, userId: string) => {
+  const { location, navigator } = window;
+  const { hostname, pathname } = location;
+
+  // store the current path
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/WebAnalytics/helper.ts:26]: We should have unit tests for this file

File: openmetadata-ui/src/main/resources/ui/src/WebAnalytics/WebAnalyticsProvider.test.tsx (hunk 3 above)
Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/WebAnalytics/WebAnalyticsProvider.test.tsx:32]: Just the basics if it's rendering the children or not.

File: openmetadata-ui/src/main/resources/ui/src/WebAnalytics/WebAnalyticsProvider.tsx (hunk 4 above)
Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/WebAnalytics/WebAnalyticsProvider.tsx:18]: Sure.

File: openmetadata-ui/src/main/resources/ui/src/WebAnalytics/helper.ts (hunk 5 above)
Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/WebAnalytics/helper.ts:26]: Sure, I will add that.

**Comment by Sachin-chaurasiya** [comment]
Thanks @chirag-madlani , All comments are addressed.

**Review by chirag-madlani — APPROVED** [review]

File: openmetadata-ui/src/main/resources/ui/src/utils/WebAnalyticsUtils.ts [hunk 6]
```diff
@@ -12,7 +12,7 @@
  */
 
 import Analytics, { AnalyticsInstance } from 'analytics';
-import { WebPageData } from '../WebAnalytics/Webanalytics.interface';
+import { WebPageData } from '../WebAnalytics/webanalytics.interface';
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/utils/WebAnalyticsUtils.ts:15]: File name must be in PascalCase

Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/utils/WebAnalyticsUtils.ts]: @chirag-madlani , Done.

**Review by ShaileshParmar11 — APPROVED** [review]

**Review by chirag-madlani — APPROVED** [review]

**Review by ShaileshParmar11 — APPROVED** [review]

**Pull request merged**